src/QsCompiler/Transformations/Utilities.cs

```diff
@@ -55,6 +55,30 @@ public static class Utilities
                 QsNullable<Tuple<QsPositionInfo, QsPositionInfo>>.Null
             );
 
+        // All of the resolved types of the given expression must match.
+        public static TypedExpression CreateValueArray(params TypedExpression[] expressions)
```

Comment:
All of this methods are really not general, and should not be public.

Reply:
Let's discuss offline.